feat: wrong data notification from wallet instance to vci (#857)
italia/eidas-it-wallet-docs: docs/en/credential-issuance-endpoint.rst

    * - *504 Gateway Timeout* [OPTIONAL]
      - `-`
      - The Credential Issuer cannot fulfill the request within the defined time interval.

.. _notification-data-correction:

Data Correction using credential_failure
........................................

> According to `OpenID4VCI`_ Section 11, in all other unsuccessful cases ``event`` MUST use ``credential_deleted`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.

Reviewer: ```suggestion
According to `OpenID4VCI`_ Section 11, in all other unsuccessful cases ``event`` MUST use ``credential_failure`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.
```

> According to `OpenID4VCI Section 11 <https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-11>`_, in all other unsuccessful cases ``event`` MUST use ``credential_failure`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.

Author: ```suggestion
According to `OpenID4VCI Section 11 <https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-11.1>`_, in all other unsuccessful cases ``event`` MUST use ``credential_failure`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.
```

> According to `OpenID4VCI Section 11 <https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-11>`_, in all other unsuccessful cases ``event`` MUST use ``credential_failure`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.

Reviewer: ```suggestion
According to `OpenID4VCI Section 11 <https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-11>`_, in all other unsuccessful cases ``event`` MUST use ``credential_deleted`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.
```

Reviewer: https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-notification-request, *credential_deleted* is used when the unsuccessful Credential issuance was caused by a user action.

> According to `OpenID4VCI Section 11 <https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-11>`_, in all other unsuccessful cases ``event`` MUST use ``credential_deleted`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.

Author: ```suggestion
According to `OpenID4VCI`_ Section 11, in all other unsuccessful cases ``event`` MUST use ``credential_deleted`` and additional Notification Request parameters MAY be defined and used. The Credential Issuer MUST ignore any unrecognized parameters.
```

> For holder-initiated data correction, the Wallet Instance SHOULD send a Notification Request with ``event=credential_deleted`` and include additional parameters to signal the data correction context. At minimum:

Author: @m-basili may we ask why `credential_failure` is not appropriate and `credential_delete` is suggested by your team?

or it was simply a typo?

> For holder-initiated data correction, the Wallet Instance SHOULD send a Notification Request with ``event=credential_deleted`` and include additional parameters to signal the data correction context. At minimum:

Reviewer: **credential_deleted** is expected by OpenID4VCI: https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-notification-request: "_**credential_deleted** is to be used when the unsuccessful Credential issuance was caused by a user action_"

> For holder-initiated data correction, the Wallet Instance SHOULD send a Notification Request with ``event=credential_failure`` and include additional parameters to signal the data correction context. At minimum:

Reviewer: ```suggestion
For holder-initiated data correction, the Wallet Instance SHOULD send a Notification Request with ``event=credential_deleted`` and include additional parameters to signal the data correction context. At minimum:
```

- ``event_description``: concise human-readable description of the discrepancy.
- ``failure_reason`` (OPTIONAL): a short machine-readable code, e.g., ``data_correction_requested``.
- ``correction_details`` (OPTIONAL): object with minimal fields indicating impacted attributes without sensitive values (e.g., attribute identifiers only).

> Upon receipt, the Credential Issuer MAY verify against Authentic Sources and, if confirmed, SHOULD proceed with the :ref:`credential-issuance-low-level:Re-Issuance Flow`.

Reviewer 2: If the data is incorrect, I don't believe there are automatic correction mechanisms that allow for an immediate re-issuance. What are we proposing here?

Author: there an ingoing interest in mvoing this conversation to the PDND framework purposes

in this case, for all about it-wallet, the interop stops between the holder and the vci; while for data discrepancy a pdnd template should considered and beyound the specific purposes of it-wallet

we are facing the evidence that we should promote a single and consolidated way to communicate, in a structured and normalized way, the ticket of notification of data misalignment, sent from the fruitore to the erogatore

regarding your specific question, I would propose a slight change in the text to avoid any ambiguity in how the end result would appear

> Upon receipt, the Credential Issuer MAY verify against Authentic Sources and, if confirmed, SHOULD proceed with the :ref:`credential-issuance-low-level:Re-Issuance Flow`.

Author: ```suggestion
In the case of receiving a Credential failure that includes a stated reason, the Credential Issuer SHOULD forward the User's message containing that reason to the Authentic Source. It is RECOMMENDED to use PDND specific endpoints, provided by Authentic Source, to facilitate this data exchange.
```

Author: @grausof if this suggestion works also for you, I would produce the same alignment in the italian version before the merge

I would ask you to anticipate the approval of this PR to speed up the merge
